src/server_state.cpp

```diff
@@ -234,9 +240,9 @@ static int rollback_fragment(wsrep::server_state& server_state,
 
     if (!ret)
     {
-        discard_streaming_applier(server_state, high_priority_service,
-                                  streaming_applier, ws_meta);
-
+        server_state.server_service().release_high_priority_service(
```

Comment:
If we don't get here, we end up with stopped applier. Is it OK?

Reply:
The SR applier is started when the transaction starts and is stopped when the transaction commits or rolls back.

For commit, we call `discard_streaming_applier()` in the end to stop the streaming_applier in case of a succesful commit.

For rollback, we were first calling `rollback_fragment()` to roll back the transaction and then we were stopping the streaming applier. But as part of the current fix, we first stop the streaming_applier and then call `rollback_fragment()` to rollback the transaction.

So, it is fine to have the streaming applier stopped because it will be started again for the next transaction.

Reply:
In my opinion we will end up with memory leak (streaming applier not deleted) if we skip this. (err != 0)
It is already removed from streaming appliers map. For new transaction new streaming applier will be created.

We have method close_orphaned_sr_transactions() which was able to delete orphaned SR objects, but as we already removed it from the map we will end up with memory leak.